File: frontend/src/types/agent.ts
import type { PersonalityTrait } from '@/config/agent-options'

/** Agent setup configuration (before experiment starts) */
export interface AgentConfig {
  id: string
  name: string
  characterId: string
  personality: PersonalityTrait[]
  secretGoal: string
  llmModel: string
}

/** Agent status during simulation */
export type AgentStatus = 'idle' | 'thinking' | 'talking' | 'moving' | 'working' | 'sneaking' | 'exiled'

/** Trust relationship between two agents */
export interface AgentRelationship {
  trust: number // -100 to 100
  notes: string
}

/** Agent personality descriptor */
export interface AgentPersonality {
  traits: PersonalityTrait[]
  description: string
}

/** Full agent state during experiment (matches shared/schemas/agent.json) */
export interface Agent {
  id: string
  name: string
  characterId: string
  personality: AgentPersonality
  secretGoal: string
  llmModel: string
  location: string
  status: AgentStatus
  suspicionLevel: number // 0-100
  inventory: string[]
  relationships: Record<string, AgentRelationship>
}

File: backend/app/schemas/agent.py
from uuid import UUID

from pydantic import Field

from app.db.models import AgentStatus
from app.agents.models import AgentMemoryState, PersonalityAxes, SecretGoal
from app.schemas.common import APIModel


class Personality(APIModel):
    axes: PersonalityAxes
    trait_tags: list[str] = Field(default_factory=list)
    self_concept: str | None = None


class Relationship(APIModel):
    trust: float | None = Field(default=None, ge=-100, le=100)
    notes: str | None = None


class AgentRead(APIModel):
    id: UUID
    name: str
    character_id: str | None = None
    personality: Personality
    secret_goal: str
    goal_archetype: str | None = None
    goal: SecretGoal | None = None
    llm_model: str
    location: str | None = None
    status: AgentStatus = AgentStatus.IDLE
    suspicion_level: float = Field(default=0, ge=0, le=100)
    inventory: list[str] = Field(default_factory=list)
    memory: AgentMemoryState | None = None
    relationships: dict[str, Relationship] = Field(default_factory=dict)
